### ISSUE TITLE
Unnecessary Apply rule being applied for cases when apply is needed

### ISSUE THREAD

**Issue by jck2** [detekt/detekt#3561]
Hi,

I found discovered an issue regarding adding the UnnecessaryApply rule. 

It appears that this rule correctly covers the most basic cases where an apply block is used for only one statement such as the following:

`textview.apply { visibility = View.VISIBLE }`.

However, I noticed this is also flagging cases like the following, where the apply actually is needed multiple times but is only used in that manner in a nested block:

```
textview.apply {
    post {
        visibility = View.VISIBLE
        text = "Hello world!"
    }
}
```

The block above is flagged under this rule but the apply is actually being used inside the post block.

Looking at the commits where this rule was added: https://github.com/detekt/detekt/pull/1229, the rule seems to be only checking if the body of the outer lambda has one statement, but doesn't inspect anything nested within that statement

**Comment by BraisGabin** [comment]
This doesn't seem like a false positive to me. You should use `let` instead of `apply` in this case.

The usage of `apply` is only encoraged if you are configuring an instance and then you get the instance. Example:

```kotlin
val paint = Paint().apply { color = RED }
```

In any other case the usage of `apply` is discoraged because you are changing `this` and a lot of issues can happen.

I know that we had this conversation before but I can't find the issue :(

**Comment by BraisGabin** [comment]
Or, if you don't want to use the `it.` or name the parameter you can use `with`:

```kotlin
with(textview) {
    post {
        visibility = View.VISIBLE
        text = "Hello world!"
    }
}
```

And these are not worksrounds to the rule. They are actual fixes.
